### PR TITLE
Reduce allocations in CSharpCommandLineParse

### PR DESCRIPTION
**Pull request by davkean**
Remove a couple of unneeded allocations in CSharpCommandLineParser.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs [hunk 1]
```diff
@@ -1668,16 +1668,16 @@ private static IEnumerable<CommandLineReference> ParseAssemblyReferences(string
                 alias = null;
             }
 
-            List<string> paths = ParseSeparatedPaths(value).Where((path) => !string.IsNullOrWhiteSpace(path)).ToList();
+            string[] paths = ParseSeparatedPaths(value).Where((path) => !string.IsNullOrWhiteSpace(path)).ToArray();
```
Comment (Therzok) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: I'd say `ToList()` is better than `ToArray()` here, in that it avoids the last allocation caused by trimming the collection to so `Capacity == Length`

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs [hunk 2]
```diff
@@ -1618,7 +1618,7 @@ private static IEnumerable<CommandLineAnalyzerReference> ParseAnalyzers(string a
                 yield break;
             }
 
-            List<string> paths = ParseSeparatedPaths(value).Where((path) => !string.IsNullOrWhiteSpace(path)).ToList();
+            IEnumerable<string> paths = ParseSeparatedPaths(value).Where((path) => !string.IsNullOrWhiteSpace(path));
```
Comment (sharwell) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: 💭 This one has the downside of making it harder to see the result while debugging. Is it really that big of a win?

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: Agree. In general ToList is better than ToArray from an allocation perspective. Is there a subtle issue we're missing here?

**Comment by jaredpar** [comment]
@Pilchie still waiting on why ToArray vs. ToList coment

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs (hunk 1 above)
Comment (davkean) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: Yeah, I looked at this again - you are right. I'll back this out.

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs (hunk 2 above)
Comment (davkean) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: This is producing about 100 KB of garbage opening Roslyn. Every bit counts, don't buy the debugging argument that's what results view is for.

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs (hunk 1 above)
Comment (jaredpar) [reply src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: @davkean can you make this change so we can merge?

**Comment by jaredpar** [comment]
Closing due to lack of activity.

**Pull request closed without merging**